lib/http.c

```diff
@@ -4454,15 +4453,57 @@ static struct name_const H2_NON_FIELD[] = {
   { STRCONST("Transfer-Encoding") },
 };
 
-static bool h2_non_field(const char *name, size_t namelen)
+static bool h2_permissible_field(struct dynhds_entry *e)
 {
   size_t i;
   for(i = 0; i < CURL_ARRAYSIZE(H2_NON_FIELD); ++i) {
-    if(namelen < H2_NON_FIELD[i].namelen)
+    if(e->namelen < H2_NON_FIELD[i].namelen)
+      return TRUE;
+    if(e->namelen == H2_NON_FIELD[i].namelen &&
+       strcasecompare(H2_NON_FIELD[i].name, e->name))
       return FALSE;
-    if(namelen == H2_NON_FIELD[i].namelen &&
-       strcasecompare(H2_NON_FIELD[i].name, name))
+  }
+  return TRUE;
+}
+
+static const char *http_skip_quoted_string(const char *s)
+{
+  DEBUGASSERT(*s == '"');
+  if(!*s)
+    return s;
+  s++;
+  while(*s && *s != '"')
+    s++;
+  if(*s == '"')
+    s++;
+  return s;
+}
```

Comment:
Maybe you can use `Curl_str_quotedword` instead? It is almost the same function.

Reply:
Thanks for the hint. Added.